pkg/metadata/image_metadata.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"encoding/json"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/metadata/store"
+
+	runtime "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+// The code is very similar to sandbox.go, but there is no template support
+// in golang, thus similar files for different types.
+// TODO(random-liu): Figure out a way to simplify this.
+// TODO(random-liu): Handle versioning
+
+// imageMetadataVersion is current version of image metadata.
+const imageMetadataVersion = "v1" // nolint
+
+// versionedImageMetadata is the internal struct representing the versioned
+// image metadata
+// nolint
+type versionedImageMetadata struct {
+	// Version indicates the version of the versioned image metadata.
+	Version string `json:"version,omitempty"`
+	ImageMetadata
+}
+
+// ImageMetadata is the unversioned image metadata.
+type ImageMetadata struct {
+	// Id of the image. Normally the Digest
+	ID string `json:"id,omitempty"`
+	// Other names by which this image is known.
+	RepoTags []string `json:"repo_tags,omitempty"`
+	// Digests by which this image is known.
+	RepoDigests []string `json:"repo_digests,omitempty"`
```

Comment:
I would highly suggest that you not export the concept of `RepoTags` or `RepoDigests` into CRI. The concept, from a data modeling perspective, is extremely broken and was an unfortunate addition to docker.

Is there anyway we can stop this from propagating?